crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -175,26 +188,46 @@ impl RevenueRecoveryPaymentsAttemptStatus {
                     );
                 };
 
-                // get a reschedule time
-                let schedule_time = get_schedule_time_to_retry_mit_payments(
-                    db,
-                    &revenue_recovery_payment_data
-                        .merchant_account
-                        .get_id()
-                        .clone(),
-                    process_tracker.retry_count + 1,
+                // Reopen calculate workflow on payment failure
+                reopen_calculate_workflow_on_payment_failure(
+                    state,
+                    &process_tracker,
+                    payment_intent,
+                    revenue_recovery_payment_data,
                 )
-                .await;
+                .await?;
+                // get a reschedule time
```

Comment:
Why these code where commented if not required remove those commented lines, other places also